src/app/components/Document/index.jsx

```diff
@@ -3,16 +3,49 @@ import ResourceHints from './ResourceHints';
 import '../../lib/globalStyles';
 import { C_POSTBOX } from '../../lib/constants/styles';
 
+const isAmpPayload = htmlAttributes =>
+  htmlAttributes.toString().includes('amp');
+
+/* eslint-disable react/no-danger */
+/* disabling the rule that bans the use of dangerouslySetInnerHTML until a more appropriate implementation can be implemented */
+const injectClientSideApp = serialisedData => {
+  if (serialisedData) {
+    return (
+      <script
+        dangerouslySetInnerHTML={{
+          __html: `window.SIMORGH_DATA=${serialisedData}`,
+        }}
+      />
+    );
+  }
+
+  return null;
+};
+
 /* eslint-disable react/prop-types */
 const Document = ({ assets, app, data, styleTags, helmet }) => {
   const htmlAttrs = helmet.htmlAttributes.toComponent();
   const meta = helmet.meta.toComponent();
   const title = helmet.title.toComponent();
   const links = helmet.link.toComponent();
-  const serialisedData = JSON.stringify(data);
-  const scripts = assets.map(asset => (
-    <script key={asset} type="text/javascript" src={asset} defer />
-  ));
+  let scripts = [];
+  let ampScript;
+  let serialisedData;
+
+  if (isAmpPayload(helmet.htmlAttributes)) {
```

Comment:
I've added the logic to handle the injection of the amp `<script>` and the removal of other `<script>`'s here in Document as it is currently an isolated use case. In the future it is likely that we will need to inject different scripts for different use cases, at this point it may be more beneficial to abstract the logic into a `Scripts` component similarly to how we handle `MetaData` within an isolated component.

Reply:
We should just handle injection of the AMP script in the [MetaData component](https://github.com/bbc/simorgh/blob/latest/src/app/components/Metadata/index.jsx). This component already has access to a boolean of `amp` so the logic is simple to add.

Would just need to extract `scripts` out of react-helmet and inject it into the document.